internal/converter/internal/staticconvert/internal/build/builder.go

```diff
@@ -367,6 +373,45 @@ func (b *IntegrationsConfigBuilder) appendExporterV2(commonConfig *common_v2.Met
 	b.diags.AddAll(prometheusconvert.AppendAllNested(b.f, promConfig, jobNameToCompLabelsFunc, extraTargets, remoteWriteExports))
 }
 
+func (b *ConfigBuilder) appendTraces() {
+	if reflect.DeepEqual(b.cfg.Traces, traces.Config{}) {
+		return
+	}
+
+	for _, cfg := range b.cfg.Traces.Configs {
+		otelCfg, err := cfg.OtelConfig()
+		removePushReceiver(otelCfg)
+
+		if err != nil {
+			b.diags.Add(diag.SeverityLevelCritical, fmt.Sprintf("failed to load otelConfig from agent traces config: %s", err))
+			continue
+		}
+		b.diags.AddAll(otelcolconvert.AppendConfig(b.f, otelCfg))
+	}
+}
+
+// removePushReceiver removes the push_receiver from the otel config. The
+// push_receiver is an implementation detail for static traces that is not
+// necessary for the flow configuration.
+func removePushReceiver(otelCfg *otelcol.Config) {
+	if _, ok := otelCfg.Receivers[otel_component.NewID("push_receiver")]; !ok {
+		return
+	}
+
+	delete(otelCfg.Receivers, otel_component.NewID("push_receiver"))
+	spr := make([]otel_component.ID, len(otelCfg.Service.Pipelines[otel_component.NewID("traces")].Receivers)-1)
+	offset := 0
+	for i, r := range otelCfg.Service.Pipelines[otel_component.NewID("traces")].Receivers {
+		if r == otel_component.NewID("push_receiver") {
+			offset = 1
+			continue
+		}
+
+		spr[i-offset] = r
+	}
+	otelCfg.Service.Pipelines[otel_component.NewID("traces")].Receivers = spr
```

Comment:
I find the logic with the offset a bit hard to follow and visibly prove to be correct. Have you considered creating a slice with 0 length but the appropriate capacity so you could use appends instead? 

```go 
tracesID := otel_component.NewID("traces") 
spr := make([]otel_component.ID, 0, len(otelCfg.Service.Pipelines[tracesID].Receivers) - 1) 

for _, r := range otelCfg.Service.Pipelines[tracesID].Receivers {
    if r == otel_component.NewID("push_receiver") {
        continue 
    }    

    spr = append(spr, r)  
}

otelCfg.Service.Pipelines[tracesID].Receivers = spr 
```

Reply:
ya I'll give that a go for clarity